plugins/get_url/komand_get_url/triggers/poll_file/trigger.py

```diff
@@ -14,65 +16,47 @@ def __init__(self):
-
-            """Download file"""
-            url_object = komand.helper.open_url(
-                url,
-                verify=is_verify,
-                If_None_Match=headers.get("etag", ""),
-                If_Modified_Since=headers.get("last-modified", ""),
-                User_Agent=params.get(Input.USER_AGENT, "Mozilla/5.0"),
-            )
-
-            """File modified"""
+            url_object, meta = self.utils.check_prefix_and_download(url, is_verify, user_agent)
+            # File modified
             if url_object:
-                contents = url_object.read().decode("utf-8")
-
-                """Write etag and last modified to cache"""
-                utils.create_url_meta_file(meta, url_object)
-
-                """We can't guarantee server supports lastmodified/etag, compare contents"""
-                if komand.helper.check_cachefile(cache_file):
-                    old = komand.helper.open_cachefile(cache_file)
-                    old_contents = old.read()
-                    old.close()
-                    if old_contents == contents:
-                        is_modified = False
-                        self.logger.info("GetUrl: File not updated")
-
-                """Write new URL file contents to cache"""
-                if is_modified:
-                    f = komand.helper.open_cachefile(cache_file)
-                    f.write(contents)
-                    f.close()
-
-                    """Check URL status code and return file contents"""
-                    if url_object.code >= 200 or url_object.code <= 299:
-                        f = komand.helper.encode_string(contents)
-                        if f:
-                            output = {
-                                Output.BYTES: f.decode("utf-8"),
-                                Output.STATUS_CODE: url_object.code or 200,
-                            }
-                            self.send(output)
-
+                self._save_to_cache_and_send(url_object, meta)
             time.sleep(poll)
+
+    def _save_to_cache_and_send(self, url_object, meta):
+        cache_file = constants.DEFAULT_CACHE_FOLDER + meta.get("file")
+        contents = url_object.read().decode(constants.DEFAULT_ENCODING, "replace")
+
+        # Write etag and last modified to cache
+        self.utils.create_url_meta_file(meta, url_object)
+
+        # We can't guarantee server supports lastmodified/etag, compare contents
+        if komand.helper.check_cachefile(cache_file):
+            old_cache_file = komand.helper.open_cachefile(cache_file)
+            old_contents = old_cache_file.read()
+            old_cache_file.close()
+            if old_contents == contents:
+                self.is_modified = False
+                self.logger.debug("GetUrl: File not updated")
+
+        if self.is_modified:
+            opened_cache_file = komand.helper.open_cachefile(cache_file)
+            opened_cache_file.write(contents)
```

Comment:
I'd add some exception handling here, customer might run out of memory and wouldnt know why does that fail.